manager/dispatcher/dispatcher_test.go

```diff
@@ -1141,9 +1223,18 @@ func makeTasksAndDependencies(t *testing.T, nodeID string) ([]*api.Secret, []*ap
 				Data: []byte(fmt.Sprintf("config%d", i)),
 			},
 		})
+
+		resourceRefs = append(resourceRefs, &api.ResourceReference{
+			ResourceID:   fmt.Sprintf("IDresourceRefSecret%d", i),
+			ResourceType: api.ResourceType_SECRET,
+		}, &api.ResourceReference{
+			ResourceID:   fmt.Sprintf("IDresourceRefConfig%d", i),
+			ResourceType: api.ResourceType_CONFIG,
+		})
 	}
+
 	for i, taskState := range taskStatesInOrder {
-		spec := taskSpecFromDependencies(secrets[i], secrets[len(secrets)-1], configs[i], configs[len(configs)-1])
+		spec := taskSpecFromDependencies(secrets[i], secrets[len(secrets)-1], configs[i], configs[len(configs)-1], resourceRefs[2*i], resourceRefs[2*i+1])
```

Comment:
Definitely agree.  I've added these cases as tasks to the top level of the PR and will update upon completion.